src/components/Tabs.js

```diff
@@ -5,34 +5,52 @@ class Tabs extends Component {
         super();
 
         this.state = {
+            defaultTab: null,
             selectedTab: null
         };
     }
 
     handleSelect(tab) {
         this.setState({
+            ...this.state,
```

Comment:
there is no need for this line as `this.setState` will do the merge of previous state with the new one. So it's enough to call it only with the values that have changed